lib/fibonacci.rb

```diff
@@ -1,8 +1,14 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n) - depending on n 
+# Space Complexity - O(n)  (should be O(n))
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This is the brute-force solution an O(2<sub>n</sup>) solution.  How can you solve this using dynamic programming.